libbeat/service/service_windows.go

```diff
@@ -71,9 +74,22 @@ func ProcessWindowsControlEvents(stopCallback func()) {
 	if isInteractive {
 		run = debug.Run
 	}
+
 	err = run(os.Args[0], &beatService{})
+
 	if err != nil {
-		logp.Err("Error: %v", err)
+		if errnoErr, ok := err.(syscall.Errno); ok && errnoErr == couldNotConnect {
```

Comment:
I think you could reduce the amount of nesting by returning after handling the error.